File: 052-构建乘积数组/052-multiply.py
# -*- coding:utf-8 -*-
class Solution:
    def multiply(self, A):
        # write code here
        b=[1]*len(A)
        for i in range(1,len(A)):
            b[i]=b[i-1]*A[i-1]
        temp=1
        for i in range(len(A)-1,-1,-1):
            b[i]*=temp
            temp*=A[i]
        return b

File: 052-构建乘积数组/052-multiply.cpp
class Solution {
public:
    vector<int> multiply(const vector<int>& A) {
        vector<int> ans(A.size(),1);
        for(int i=1;i<A.size();i++){
            ans[i]=ans[i-1]*A[i-1];
        }
        int temp=1;
        for(int i=A.size()-1;i>=0;i--){
            ans[i]*=temp;
            temp*=A[i];
        }
        return ans;
    }
};
